scripts/run.sh

```diff
@@ -233,10 +232,40 @@ if [[ ${E2E} == true ]]; then
     --network-runner-grpc-endpoint="0.0.0.0:12342" \
     --avalanchego-path=${AVALANCHEGO_PATH} \
     --avalanchego-plugin-dir=${AVALANCHEGO_PLUGIN_DIR} \
+    --vm-genesis-path=$BASEDIR/genesis.json \
     --output-path=$BASEDIR/avalanchego-${VERSION}/output.yaml \
     --mode=${MODE}
+}
+
+run_simulator() {
+  #################################
+  echo "building simulator"
+  pushd ./cmd/simulator
+  # to use the latest subnet-evm from the local repository
+  go mod edit -replace github.com/ava-labs/subnet-evm=../..
+  go install -v .
+  popd
+
+  echo "running simulator"
+  simulator \
+  --network-runner-output=$BASEDIR/avalanchego-${VERSION}/output.yaml \
+  --keys=./cmd/simulator/.simulator/keys \
+  --timeout=30s \
+  --concurrency=10 \
+  --base-fee=25 \
+  --priority-fee=1
+}
+
+if [[ ${MODE} == "test" || ${SIMULATOR} == true ]]; then
+  run_ginkgo
 
+  if [[ ${SIMULATOR} == true ]]; then
+    run_simulator
+  fi
```

Comment:
What are the modes of operation that we want here?

It seems like we want to run just the ginkgo tests, and if the simulator is enabled we want to run the ginkgo tests and then run the simulator as well? I'm confused by the intent here since we skip a bunch of the ginkgo tests if the mode is not set to true.

Reply:
Yeah at some point, we decided to skip ginkgo for the mode `test`. This case is for `MODE=run SIMULATOR=true scripts/run.sh` where we run network runner + simulator.